fastly/user_test.go

```diff
@@ -0,0 +1,143 @@
+package fastly
+
+import "testing"
+
+func TestClient_ListUsers(t *testing.T) {
+	t.Parallel()
+
+	var users []*User
+	var err error
+	record(t, "users/list", func(c *Client) {
+		users, err = c.ListUsers()
+	})
+	if err != nil {
+		t.Fatal(err)
+	}
+	if len(users) < 1 {
+		t.Errorf("bad users: %v", users)
+	}
+}
+
+func TestClient_ListCustomerUsers(t *testing.T) {
+	t.Parallel()
+
+	var users []*User
+	var err error
+	record(t, "users/list_customer", func(c *Client) {
+		users, err = c.ListCustomerUsers(&ListCustomerUsersInput{
+			ID: "CUsT0m3rID",
+		})
+	})
+	if err != nil {
+		t.Fatal(err)
+	}
+	if len(users) < 1 {
+		t.Errorf("bad users: %v", users)
+	}
+}
+
+func TestClient_GetCurrentUser(t *testing.T) {
+	t.Parallel()
+
+	var user *User
+	var err error
+	record(t, "users/get_current_user", func(c *Client) {
+		user, err = c.GetCurrentUser()
+	})
+	if err != nil {
+		t.Fatal(err)
+	}
+	t.Logf("%+v", user)
+}
+
+func TestClient_GetUser(t *testing.T) {
+	t.Parallel()
+
+	input := &GetUserInput{
+		ID: "T3hr33eE3thr3EEtHr3e3",
+	}
+
+	var user *User
+	var err error
+	record(t, "users/get", func(c *Client) {
+		user, err = c.GetUser(input)
+	})
+	if err != nil {
+		t.Fatal(err)
+	}
+	t.Logf("%+v", user)
+}
+
+func TestClient_CreateUser(t *testing.T) {
+	t.Parallel()
+
+	input := &CreateUserInput{
+		Login: "new+engineer@example.com",
+		Name:  "new engineer",
+		Role:  "engineer",
+	}
+
+	var user *User
+	var err error
+	record(t, "users/create", func(c *Client) {
+		user, err = c.CreateUser(input)
+	})
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	if user.Login != input.Login {
+		t.Errorf("returned invalid login, got %s, want %s", user.Login, input.Login)
+	}
+
+	if user.Name != input.Name {
+		t.Errorf("returned invalid name, got %s, want %s", user.Name, input.Name)
+	}
+
+	if user.Role != input.Role {
+		t.Errorf("returned invalid role, got %s, want %s", user.Role, input.Role)
+	}
+}
+
+func TestClient_UpdateUser(t *testing.T) {
+	t.Parallel()
+
+	input := &UpdateUserInput{
+		ID:   "T3hr33eE3thr3EEtHr3e3",
+		Name: "Superuser Three",
+		Role: "superuser",
+	}
+
+	var user *User
+	var err error
+	record(t, "users/update", func(c *Client) {
+		user, err = c.UpdateUser(input)
+	})
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	if user.Name != input.Name {
+		t.Errorf("returned invalid name, got %s, want %s", user.Name, input.Name)
+	}
+
+	if user.Role != input.Role {
+		t.Errorf("returned invalid role, got %s, want %s", user.Role, input.Role)
+	}
+}
+
+func TestClient_DeleteUser(t *testing.T) {
+	t.Parallel()
+
+	input := &DeleteUserInput{
+		ID: "6SI6xsIX66S66iX6ixSIx",
+	}
+
+	var err error
+	record(t, "users/delete", func(c *Client) {
+		err = c.DeleteUser(input)
+	})
+	if err != nil {
+		t.Fatal(err)
+	}
+}
```

Comment:
These tests are not generic since you hardcoded the IDs:

```
--- FAIL: TestClient_DeleteUser (0.34s)
    /Users/pschulte/Documents/go-workspace/src/github.com/fastly/go-fastly/fastly/user_test.go:141: 404 - Not Found:
        
            Title:  Record not found
            Detail: Cannot find user '6SI6xsIX66S66iX6ixSIx'
--- FAIL: TestClient_ListCustomerUsers (0.34s)
    /Users/pschulte/Documents/go-workspace/src/github.com/fastly/go-fastly/fastly/user_test.go:32: 404 - Not Found:
        
            Title:  Record not found
            Detail: Cannot find customer 'CUsT0m3rID'
--- FAIL: TestClient_UpdateUser (0.35s)
    /Users/pschulte/Documents/go-workspace/src/github.com/fastly/go-fastly/fastly/user_test.go:117: 404 - Not Found:
        
            Title:  Record not found
            Detail: Cannot find user 'T3hr33eE3thr3EEtHr3e3'
--- FAIL: TestClient_GetUser (0.56s)
    /Users/pschulte/Documents/go-workspace/src/github.com/fastly/go-fastly/fastly/user_test.go:66: 404 - Not Found:
        
            Title:  Record not found
            Detail: Cannot find user 'T3hr33eE3thr3EEtHr3e3'
FAIL
FAIL	github.com/fastly/go-fastly/fastly	0.571s
Error: Tests failed.
```

Please create a generic test suite that doesn't need to be modified in order to pass the tests. Usually, we create one function to test the CRUD functionality and then have other functions to test the required params.

Reply:
apparently I copied the pattern in the wrong module (`token_test.go` fwiw) :-P
I've updated now